### PR TITLE
Move catalog-info.yaml and .tekton to skeleton

### PR DESCRIPTION
**Pull request by yangcao77**
The work included in this PR:

- Move `catalog-info.yaml` and `.tekton` skeletons to `skeleton/`
- create a new step in `backstage/template.yaml` to fetch skeletons
- update `import-samples` to sed substitute some values in sample's template to be used for fetching skeleton. i.e. `APPTAGS`, `CATALOG_DESCRIPTION`, `DOCKERFILE`,`BUILDCONTEXT`
- rename `poc` folder to `scripts`
- create a top-level `generate.sh` to execute & automate the template generation

With this PR, the sample template folder won't contain `catalog-info.yaml` or `.tekton/`. those will be copied & render during component creation. 
See the [spring template folder ](https://github.com/yangcao77/tssc-sample-templates/tree/move-to-skeleton/templates/devfile-sample-java-springboot-basic/content) vs the [generated spring component repo](https://github.com/stephanie-cy/spring1/tree/main/)

The following is a generated repo, and screenshots to the 4-step template output (can see with a new step `Fetch Skeleton`) and also screenshots to the catalog view: 
https://github.com/stephanie-cy/spring1/tree/main/
Screenshot 2023-12-08 at 2 05 51 PM

Screenshot 2023-12-08 at 2 05 59 PM

### CODE REVIEW

**Comment by yangcao77** [comment]
@jduimovich PTAL

File: skeleton/.tekton/docker-pull-request.yaml
Comment (jduimovich) [inline skeleton/.tekton/docker-pull-request.yaml:1]:
```
    value: sed.edit.DOCKERFILE
    value: Dockerfile
```

Pull request should also have a template var like the push template .
```
    value: sed.edit.DOCKERFILE
    value: ${{ values.dockerfile }}
```

Comment (jduimovich) [inline skeleton/.tekton/docker-pull-request.yaml:1]: I'm ok if we merge this and follow up with a small fix

**Review by jduimovich — APPROVED** [review]
Changed the one file ... added a template var but LGTM
Thanks for the quick work and sorry I didn't notice you had submitted the PR

**Pull request merged**